File: lib/src/main/java/fr/o80/sample/lib/di/LibComponent.java
package fr.o80.sample.lib.di;

import javax.inject.Singleton;

import dagger.Component;
import fr.o80.sample.lib.core.LibConfiguration;
import fr.o80.sample.lib.core.ui.LibNavigationView;
import fr.o80.sample.lib.ui.MainActivity;

/**
 * @author Olivier Perez
 */
@Component(modules = LibModule.class)
@Singleton
public interface LibComponent {
    void inject(MainActivity activity);

    void inject(LibNavigationView view);

    LibConfiguration libConfiguration();
}

File: lib/src/main/java/fr/o80/sample/lib/di/LibComponent.kt
package fr.o80.sample.lib.di

import dagger.Component
import fr.o80.crapp.data.ProjectRepository
import fr.o80.crapp.data.TimesheetRepository
import fr.o80.sample.lib.core.LibConfiguration
import fr.o80.sample.lib.core.ui.LibNavigationView
import fr.o80.sample.lib.ui.MainActivity
import javax.inject.Singleton

/**
 * @author Olivier Perez
 */
@Component(modules = [LibModule::class])
@Singleton
interface LibComponent {
    fun inject(activity: MainActivity)

    fun inject(view: LibNavigationView)

    fun libConfiguration(): LibConfiguration

    fun projectRepository(): ProjectRepository

    fun timesheetRepository(): TimesheetRepository
}
